### DISCUSSION TITLE
FAILED AUTHENTICATION

### DISCUSSION THREAD

**Discussion by Omabelle73** [community/community#52046] [New to GitHub]
Please how do I tackle authentication issues on GitHub. My code keep saying authentication failed after I "git push"

**Comment by airtower-luna** [comment]
Use a [valid personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) with **repo** and **workflow** scopes (if you're using HTTPS) or a registered SSH key (if you're using SSH) to authorize the push.

If you need more help, you'll need to be a lot more specific about the problem. A good problem description should include at least:

* What exactly are you doing (commands, etc.)?
* What happens (full error messages, etc.)?
* What would you expect to happen instead?

Try to be as precise as possible.

**Comment by SMelidoni** [comment]
Hi there @Omabelle73 👋🏽

Authentication issues can be frustrating, but don't worry! I'll guide you through some steps to resolve the "authentication failed" error when you try to `git push`. Please follow these steps:

1. **Check your remote URL**: Make sure you're using the correct remote URL for your repository. You can check the current remote URL with:

`git remote -v`

If the remote URL is incorrect, you can update it using:

`git remote set-url origin <correct-remote-url>`

Replace `<correct-remote-url>` with the correct URL for your repository.

2. **Verify your credentials**: Ensure you have the correct credentials set up for your GitHub account. You can use either HTTPS (with a personal access token) or SSH (with an SSH key) for authentication.

- For **HTTPS**, follow the [documentation on creating a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and then [configure Git to use the token](https://docs.github.com/en/get-started/getting-started-with-git/caching-your-github-credentials-in-git).

- For **SSH**, follow the [documentation on connecting to GitHub with SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh) to set up and configure your SSH keys.

3. **Confirm your permissions**: Ensure you have the appropriate permissions to push to the repository. If you're not the repository owner or a collaborator, you may not have push access. In this case, you'll need to fork the repository, push to your fork, and then create a pull request to contribute to the original repository.

If you've followed these steps and are still experiencing authentication issues, please provide more details about the error message and your configuration, and I'll be happy to help you further!

**Reply by geekrasta** [reply]
im getting error 403 still
